cluster-autoscaler/cloudprovider/baiducloud/baiducloud_auto_scaling_groups.go

```diff
@@ -55,52 +55,46 @@ func (m *autoScalingGroups) Register(asg *Asg) {
 }
 
 // FindForInstance returns AsgConfig of the given Instance
-func (m *autoScalingGroups) FindForInstance(instance *BaiducloudRef) (*Asg, error) {
+func (m *autoScalingGroups) FindForInstance(instanceID string) (*Asg, error) {
 	m.cacheMutex.Lock()
 	defer m.cacheMutex.Unlock()
-	if config, found := m.instanceToAsg[*instance]; found {
+	if config, found := m.instanceToAsg[instanceID]; found {
 		return config, nil
 	}
-	if _, found := m.instancesNotInManagedAsg[*instance]; found {
+	if _, found := m.instancesNotInManagedAsg[instanceID]; found {
 		// The instance is already known to not belong to any configured ASG
 		// Skip regenerateCache so that we won't unnecessarily call DescribeAutoScalingGroups
 		// See https://github.com/kubernetes/contrib/issues/2541
 		klog.V(4).Infof("instancesNotInManagedAsg")
 		return nil, nil
 	}
 	if err := m.regenerateCache(); err != nil {
-		return nil, fmt.Errorf("error while looking for ASG for instance %+v, error: %v", *instance, err)
+		return nil, fmt.Errorf("error while looking for ASG for instance %+v, error: %v", instanceID, err)
 	}
-	if config, found := m.instanceToAsg[*instance]; found {
+	if config, found := m.instanceToAsg[instanceID]; found {
 		return config, nil
 	}
 	// instance does not belong to any configured ASG
 	klog.V(4).Infof("Instance %+v is not in any ASG managed by CA."+
 		" CA is now memorizing the fact not to unnecessarily call BCE API afterwards trying to find the "+
-		"unexistent managed ASG for the instance", *instance)
-	m.instancesNotInManagedAsg[*instance] = struct{}{}
+		"unexistent managed ASG for the instance", instanceID)
+	m.instancesNotInManagedAsg[instanceID] = struct{}{}
 	return nil, nil
 }
 
 func (m *autoScalingGroups) regenerateCache() error {
-	newCache := make(map[BaiducloudRef]*Asg)
-
+	newCache := make(map[string]*Asg)
 	// TODO: Currently, baiducloud cloudprovider not support Multiple ASG
-	instanceList, err := m.cceClient.ListInstances(m.cloudConfig.ClusterID)
```

Comment:
Multiple ASG is supported?

Reply:
Multiple ASG is not supported currently, will be supported in the next version.